### DISCUSSION TITLE
How are cores ported from Retroarch to wasm?

### DISCUSSION THREAD

**Discussion by FusionCow** [EmulatorJS/EmulatorJS#822] [Q&A]
Just wondering how cores are ported over. Can I port any core? I've looked on the website and a lot of links lead nowhere.

Thanks

**Comment by michael-j-green** [comment]
Retroarch uses libretro, and many cores have been adapted to use libretro (links to cores and systems here: https://docs.libretro.com/guides/arcade-getting-started/)

The easy ones to port to WASM are the ones that support "emscripten" as a platform in their makefiles. These generally require little to no modifications to port across.

Cores that don't have emscripten as a platform will require far more work to port and many will likely never be able to be ported.

To compile the core, we clone the build repo (see: https://github.com/EmulatorJS/build) and add the required data to the cores.json file in the root of the repo (I like to temporarily use a minimal cores.json file containing only the core I'm trying to compile while testing), and then run the build script.

If there is a successful build, I manually copy the resultant wasm files into a clone of the EmulatorJS repo and attempt to run them.

That's pretty much a high level version of the procedure.

Is there a core you have in mind that you'd like to port?
